sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -254,3 +254,54 @@ case class Pow(left: Expression, right: Expression)
       """
   }
 }
+
+object Logarithm {
+  def apply(child: Expression): Expression = new Log(child)
+}
+
+case class Logarithm(left: Expression, right: Expression)
```

Comment:
Yeah, that is what I meant saying it wasn't worth whatever code reuse we are getting.  The other option would be to name the arguments and have `def left = base`, etc.